server/src/main/java/org/opensearch/index/translog/TranslogFooter.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.index.translog;
+
+import org.apache.lucene.codecs.CodecUtil;
+import org.apache.lucene.store.OutputStreamDataOutput;
+import org.opensearch.common.io.Channels;
+import org.opensearch.core.common.io.stream.OutputStreamStreamOutput;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.channels.FileChannel;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+
+/**
+ * Handles the writing and reading of the translog footer, which contains the checksum of the translog data.
+ *
+ * Each translog file is structured as follows:
+ *
+ * 1. Translog header
+ * 2. Translog operations
+ * 3. Translog footer
+ *
+ * The footer contains the following information:
+ *
+ * - Magic number (Uint32): A constant value that identifies the start of the footer.
+ * - Algorithm ID (Uint32): The identifier of the checksum algorithm used. Currently, this is always 0.
+ * - Checksum (Uint64): The checksum of the entire translog data, calculated using the specified algorithm.
+ */
+public class TranslogFooter {
```

Comment:
Curious will Translog Footer only apply to remote translog?

Reply:
No, the Translog footer will be applied to both Remote and Local translog. The footer will be applied at the time of generation rollover, when the translog file becomes immutable. By instantiating `TranslogCheckedContainer` for both Local and Remote translog, we will now have the checksum pre-calculated for both, which will be written into footer.

For remote translog, upon each sync, the files are transferred to remote and the generation is rolled over.
For local translog, the sync can happen independently of generation rollover. So, we will keep writing to file until the rollover happens.